assets/js/modules/adsense/index.js

```diff
@@ -47,17 +41,24 @@ import {
 import AdSenseIcon from '../../../svg/adsense.svg';
 import { STORE_NAME } from './datastore/constants';
 import { ERROR_CODE_ADBLOCKER_ACTIVE } from './constants';
+import { registerStore as registerDataStore } from './datastore';
 
 addFilter(
 	'googlesitekit.AdSenseDashboardZeroData',
 	'googlesitekit.AdSenseDashboardZeroDataRefactored',
 	fillFilterWithComponent( DashboardZeroData )
 );
 
-domReady( () => {
-	// IMPORTANT: When updating arguments here, also update the same call in
-	// `provideModuleRegistrations`.
-	Modules.registerModule(
+let isAdBlockerActive = () => {};
+
+export const registerStore = ( registry ) => {
+	registerDataStore( registry );
+	// TODO: fix hack
+	isAdBlockerActive = () => registry.select( STORE_NAME ).isAdBlockerActive();
+};
```

Comment:
I think this is okay as a _temporary workaround_, but it is rather ugly; it's the only thing left that isn't a clean change though. `isAdBlockerActive` here is used within the `checkRequirements` function in the `registerModule` call below. I think the better approach here would be to add support for something like "registry checks" which would be created like a registry control, using a function that works like `createRegistryControl`. That way it could still use the registry if needed (which I think is a reasonable dependency for such a function), or not if not and still be very WP-data like as well as consistent with the overall theme of dependency injection in this PR.

I found a need for a similar pattern with the refactored `CompatibilityChecks` since one of the checks needed access to the registry as well so I used a similar pattern, but just handled the registry part in the same way just without the added API/function/abstraction.

I'm happy to open a new issue for this as a follow-up related issue if we're in agreement about the way forward here?

Reply:
> I think the better approach here would be to add support for something like "registry checks" which would be created like a registry control, using a function that works like createRegistryControl. That way it could still use the registry if needed (which I think is a reasonable dependency for such a function), or not if not and still be very WP-data like as well as consistent with the overall theme of dependency injection in this PR.

That sounds like a good path forward to me, I like it.

Reply:
Why not pass the `registry` (i.e. `Data`) as parameter to a registered `checkRequirements`? I think that's a simpler way which is reasonable. Not sure why we would want to introduce a separate utility for that. If the function wants to use the `registry`, it can - if not, it doesn't need to.

Reply:
@felixarntz that's certainly one solution I considered. I opened #2749 to discuss and address this 👍